transforms/code/license_check/python/src/license_check_transform.py

```diff
@@ -0,0 +1,132 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+################################################################################
+
+
+import json
+from argparse import ArgumentParser, Namespace
+
+import pyarrow as pa
+from data_processing.data_access import DataAccess, DataAccessFactory
+from data_processing.transform import AbstractTableTransform, TransformConfiguration
+from data_processing.utils import TransformUtils, get_logger
+from transformer import AllowLicenseStatusTransformer, DenyLicenseStatusTransformer
+
+
+logger = get_logger(__name__)
+
+LICENSE_CHECK_PARAMS = "license_check_params"
+
+
+def _get_supported_licenses(license_file: str, data_access: DataAccess) -> list[str]:
+    logger.info(f"Getting supported licenses from file {license_file}")
+    licenses_list = None
+    try:
+        licenses_list_json, _ = data_access.get_file(license_file)
+        licenses_list = json.loads(licenses_list_json.decode("utf-8"))
+        logger.info(f"Read a list of {len(licenses_list)} licenses.")
+    except Exception as e:
+        logger.error(f"Failed to read file: {license_file} due to {e}")
+    return licenses_list
+
+
+class LicenseCheckTransform(AbstractTableTransform):
+    """It can be used to select the rows/records of data with licenses
+    matching those in the approved/deny list.
+    """
+
+    def __init__(self, config: dict):
+        super().__init__(config)
+
+        self.license_check = config.get(LICENSE_CHECK_PARAMS)
+        if not self.license_check:
+            raise ValueError(f"Invalid Argument: cannot create LicenseCheckTransform object.")
+        logger.debug(f"LICENSE_CHECK_PARAMS: {self.license_check}")
+        self.license_column = self.license_check["license_column_name"]
+        allow_no_license = self.license_check["allow_no_license"]
+        licenses = self.license_check["licenses"]
+        deny = self.license_check["deny"]
+        if not deny:
+            self.transformer = AllowLicenseStatusTransformer(
+                license_column=self.license_column,
+                allow_no_license=allow_no_license,
+                licenses=licenses,
+            )
+        else:
+            self.transformer = DenyLicenseStatusTransformer(
+                license_column=self.license_column,
+                allow_no_license=allow_no_license,
+                licenses=licenses,
+            )
+
+    def transform(self, table: pa.Table, file_name: str = None) -> tuple[list[pa.Table], dict]:
+        """
+        Transforms input tables by adding a boolean `license_status` column
+        indicating whether the license is approved/denied.
+        """
+        TransformUtils.validate_columns(table=table, required=[self.license_column])
+        new_table = self.transformer.transform(table)
+        return [new_table], {"n_files": 1}
```

Comment:
You can probably return empty metadata, since the runtime in general will be counting files and put this in metadata.json.  This was done in noop mostly as an example.  As alternative, you might want to return a dictionary keyed by license type and count.  THe latter is only a suggestion though.